test/pytest/test_repack_stream.py

```diff
@@ -0,0 +1,61 @@
+from pathlib import Path
+
+import numpy as np
+import pytest
+from tensorflow import keras
+
+from hls4ml.converters import convert_from_keras_model
+
+test_root_path = Path(__file__).parent
+
+
+def test_repack_precision():
+    inp = keras.Input(shape=(3, 3), name='inp')
+    out = keras.layers.Reshape((3, 3), name='reshape')(inp)
+    out = keras.layers.Conv1D(2, 2, name='conv')(out)
+    model = keras.Model(inp, out)
+
+    layer_conf = {
+        'inp': {'Precision': 'fixed<20,10>'},
+        'reshape': {'Precision': 'fixed<20,10>'},
+        'conv': {'Precision': 'fixed<20,10>'},
+    }
+
+    hls_config = {'Model': {'Precision': 'fixed<2,1>', 'ReuseFactor': 1}, 'LayerName': layer_conf}
+
+    # Repack only happens in io_stream
+    model_hls = convert_from_keras_model(model, hls_config=hls_config, io_type='io_stream')
+    assert 'repack_reshape' in model_hls.graph, 'repack_reshape not found in graph'
+    repack_precision = model_hls.graph['repack_reshape'].attributes['result_t'].precision
+    assert repack_precision.integer == 10, 'Precision mismatch'
+    assert repack_precision.fractional == 10, 'Precision mismatch'
+    assert repack_precision.width == 20, 'Precision mismatch'
+    assert repack_precision.signed is True, 'Precision mismatch'
+
+
+@pytest.mark.parametrize('backend', ['vivado', 'vitis', 'quartus'])
+@pytest.mark.parametrize('strategy', ['Latency', 'Resource'])
+def test_repack(backend: str, strategy: str):
+    inp1 = keras.Input(shape=(4,), name='inp1')
+    inp2 = keras.Input(shape=(4,), name='inp2')
+    r1 = keras.layers.Reshape((2, 2), name='reshape1')(inp1)
+    r2 = keras.layers.Reshape((2, 2), name='reshape2')(inp2)
+    out = keras.layers.Concatenate(name='concat')([r1, r2])
+    model = keras.Model([inp1, inp2], out)
+
+    hls_config = {'Model': {'Precision': 'ap_ufixed<8,8>', 'ReuseFactor': 1}, 'Strategy': strategy}
+    model_hls = convert_from_keras_model(
+        model,
+        io_type='io_stream',
+        backend='Vivado',
+        hls_config=hls_config,
+        output_dir=str(test_root_path / f'{backend}_{strategy}'),
```

Comment:
Generally the names have been of the style, hl4mlprj_repack_{backend}_{strategy}. The way it is currently, if you run pytest locally it's hard to refer back to the results.